### PR TITLE
Remove dotnet sdk dependency

### PR DESCRIPTION
**Pull request by embetten**
- remove dotnet sdk dependency by checking for runtimes instead

#51 #53

### CODE REVIEW

File: src/artifacts_keyring/plugin.py [hunk 1]
```diff
@@ -33,13 +33,14 @@ def __init__(self):
             self.exe = [tool_path]
         else:
             try:
-                sys_version = tuple(int(i) for i in
-                    subprocess.check_output(["dotnet", "--version"]).decode().strip().partition("-")[0].split("."))
-                get_runtime_path = lambda: "dotnet"
+                # check to see if any dotnet runtimes are installed. Not checking specific versions.
+                output = subprocess.check_output(["dotnet", "--list-runtimes"]).decode().strip()
```
[the remark this answers is not in the document]
Comment (embetten) [reply src/artifacts_keyring/plugin.py:37]: The dotnet requirement is from the underlying [artifacts cred provider](https://github.com/microsoft/artifacts-credprovider) that this code invokes. I was going to let that cred provider settle the actual runtime version.

File: src/artifacts_keyring/plugin.py [hunk 2]
```diff
@@ -33,13 +33,14 @@ def __init__(self):
...
+                if(len(output) == 0):
+                    raise Exception("No dotnet runtime found.")
```
[the remark this answers is not in the document]
Comment (embetten) [reply src/artifacts_keyring/plugin.py]: done

File: src/artifacts_keyring/plugin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cormacpayne) [reply src/artifacts_keyring/plugin.py:37]: OK makes sense, thanks for the explanation.

**Review by phil-hodgson — APPROVED** [review]
LGTM

**Pull request merged**